assets/js/pages/SapSystemsOverviewPage/SapSystemsOverview.test.jsx

```diff
@@ -80,13 +84,93 @@ describe('SapSystemsOverviews component', () => {
         tenant
       );
       expect(mainRow.querySelector('td:nth-child(5)')).toHaveTextContent(
-        dbAddress
+        sapSystemType
       );
       expect(mainRow.querySelector('td:nth-child(6)')).toHaveTextContent(
+        dbAddress
+      );
+      expect(mainRow.querySelector('td:nth-child(7)')).toHaveTextContent(
         'ENSA1'
       );
     });
 
+    it('should display the correct SAP system type', () => {
+      const sapSystemTypes = [
+        'ABAP',
+        'J2EE',
+        'ABAP|J2EE',
+        'J2EE|ABAP',
+        'SOME_SAP_SYSTEM_FEATURE|NOT_A_REAL_SYSTEM',
+      ];
+
+      const expectedSapSystemTypes = [
+        'ABAP',
+        'JAVA',
+        'ABAP/JAVA',
+        'ABAP/JAVA',
+        '',
+      ];
+      const sapSystems = [
+        sapSystemFactory.build({
+          sap_system_type: sapSystemTypes[0],
+        }),
+        sapSystemFactory.build({
+          sap_system_type: sapSystemTypes[1],
+        }),
+        sapSystemFactory.build({
+          sap_system_type: sapSystemTypes[2],
+        }),
+        sapSystemFactory.build({
+          sap_system_type: sapSystemTypes[3],
+        }),
+        sapSystemFactory.build({
+          sap_system_type: sapSystemTypes[4],
+        }),
+      ];
+
+      const sapSystemApplicationInstances = [
+        sapSystems[0].application_instances,
+        sapSystems[1].application_instances,
+        sapSystems[2].application_instances,
+        sapSystems[3].application_instances,
+        sapSystems[4].application_instances,
+      ].flat();
+
+      renderWithRouter(
+        <SapSystemsOverview
+          sapSystems={sapSystems}
+          userAbilities={userAbilities}
+          applicationInstances={sapSystemApplicationInstances}
+          databaseInstances={[]}
+        />
+      );
+      const rows = screen.getByRole('table').querySelectorAll('tbody > tr');
+      const firstSapSystem = rows[0];
+      const secondSapSystem = rows[2];
+      const thirdSapSystem = rows[4];
+      const fourthSapSystem = rows[6];
+      const fifthSapSystem = rows[8];
+
+      expect(firstSapSystem.querySelector('td:nth-child(5)')).toHaveTextContent(
```

Comment:
Could we have this `expect`s in some sort of loop.
I mean, you are doing the same operation 5 times in a row, just suming the index twice or so

Reply:
Refactored with a for each!